server/src/main/java/org/opensearch/index/cache/query/QueryCacheStats.java

```diff
@@ -77,6 +77,9 @@ public QueryCacheStats(long ramBytesUsed, long hitCount, long missCount, long ca
     }
 
     public void add(QueryCacheStats stats) {
+        if (stats == null) {
+            return;
+        }
```

Comment:
This can be null when query cache stats are not requested.  There were null pointer exception when computing AggregatedNodeLevelStats. For other stats, null check is already present.

Reply:
Lets add the null check at the caller only